subsys/bluetooth/controller/ll_sw/isoal.c

```
@@ -0,0 +1,414 @@
+	 *   4) [         #########          ] : We lack the beginning. Not yet completed, sdu lost
+	 *   5) [#########          #########] : We lack something. Completed. Frame is lost.
+	 *   6) [                            ] : Predict
+	 */
+
+	const uint8_t llid = isoal_rx_packet_classify(pckt_meta);
+
+	/* TBD Below case statement is very redundant, might simplify when
+	 * development has progressed further
+	 */
+	switch (sink->sdu_production.fsm) {
+
+	case ISOAL_START:
+		/* State assumes First PDU of new SDU */
+		sink->sdu_production.prev_pdu_id = pckt_meta->meta->id;
+		if (llid == ISOAL_PACKET_START_OR_CONTINUATION) {
+			/* PDU contains first fragment of an SDU */
+			err |= isoal_rx_append_to_sdu(sink, pckt_meta, false);
+			sink->sdu_production.fsm = ISOAL_CONTINUE;
+		} else if (llid == ISOAL_PACKET_END_OR_COMPLETE) {
+			/* PDU contains complete SDU */
+			err |= isoal_rx_append_to_sdu(sink, pckt_meta, true);
+			sink->sdu_production.fsm = ISOAL_START;
+		} else {
+			/* Unsupported case */
+
+			/* TBD error handling */
+			break;
+		}
+		break;
+
+	case ISOAL_CONTINUE:
+		/* State assumes that at least one PDU has been seen of fragmented SDU */
+		id = pckt_meta->meta->id;
+		prev_id = sink->sdu_production.prev_pdu_id;
+		if (id != (prev_id+1)) {
+			/* Id not as expexted, a PDU could have been lost */
+
+			/* TBD error handling? Mark SDU as incomplete? break? */
+		}
+		sink->sdu_production.prev_pdu_id = id;
+		if (llid == ISOAL_PACKET_START_OR_CONTINUATION) {
+			/* PDU contains a contiuation (neither start of end) fragment of SDU */
+			err |= isoal_rx_append_to_sdu(sink, pckt_meta, false);
+			sink->sdu_production.fsm = ISOAL_CONTINUE;
+		} else if (llid == ISOAL_PACKET_END_OR_COMPLETE) {
+			/* PDU contains end fragment of a fragmented SDU */
+			err |= isoal_rx_append_to_sdu(sink, pckt_meta, true);
+			sink->sdu_production.fsm = ISOAL_START;
+		} else  {
+			/* Unsupported case */
+
+			/* TBD error handling */
```

Maybe assert instead of silently ignoring the problem?

---

Done